src/contracts/TopChef.sol

```diff
@@ -115,47 +119,32 @@ contract TopChef is Chef {
         }
     }
 
-    function harvest(uint256 agIndex) public {
+    function harvest(uint256 agIndex) public returns (uint256) {
         // Checks
         if (!(areRewardsActive())) revert RewardsInactive();
-        if ((viewPendingHarvest(agIndex) == 0)) revert NoRewardsToClaim();
         AllocationGroup storage group = _allocations[agIndex];
         // TODO do we want a backup in case a group looses access to their wallet
-        if (!(group.groupAddress == _msgSender() || _msgSender() == owner())) revert SenderNotGroupOrOwner();
 
         // Effects
         updateAccumulatedAllocations();
-
-        uint256 toClaim = (group.shares * getLifetimeShareValue()) / ACC_METRIC_PRECISION - group.rewardDebt;
+        uint256 toClaim = ((group.shares * (getLifetimeShareValue())) / ACC_METRIC_PRECISION) - group.rewardDebt;
 
         group.rewardDebt = group.rewardDebt + toClaim;
+        uint256 totalClaimable = group.claimable + toClaim;
+        group.claimable = totalClaimable;
 
```

Comment:
removed autoDistribute - as it's really not needed anymore - right not the rest of it is dead code - we should create a ticket to remove all autoDistribute